prelude/__internal__/Wish.mad

```diff
@@ -41,10 +47,17 @@ export chain = (f, m) => (
   ))
 )
 
-
-bad :: e -> Wish e a
-export bad = (e) => (
-  Wish((bad, good) => (bad(e)))
+chainRej :: (a -> Wish f b) -> Wish e a -> Wish f b
+export chainRej = (f, m) => (
+  Wish((bad, good) => (
+    where(m) {
+      is Wish run: run((x) => (
+        where(f(x)) {
+          is Wish run: run(bad, good)
+        }
+      ), good)
+    }
+  ))
```

Comment:
Syntactically / scan-wise this is odd to me with the function invocation within the `where`, the boilerplate around the `where(m) { is Wish run: run` feels kinda heavy — is `m.run(` with a direct invocation not valid? Or is `where` basically our monadic accessor (hence the nesting) ? None of this is a blocker, just curious about what is valid and what you envision the language having in the future.

Reply:
Thanks to type aliases I now rewrote the implementation and a Wish became a simple binary function, implementation got a lot simpler. But otherwise that's mostly how algebraic data types and where syntax work.